src/view_control.cpp

```diff
@@ -1549,6 +1557,128 @@ void View::DrawFermata(DeviceContext *dc, Fermata *fermata, Measure *measure, Sy
     dc->EndGraphic(fermata, this);
 }
 
+void View::DrawGliss(DeviceContext *dc, Gliss *gliss, int x1, int x2, Staff *staff, char spanningType, Object *graphic)
+{
+    assert(dc);
+    assert(gliss);
+    assert(staff);
+
+    int y1 = staff->GetDrawingY();
+    int y2 = staff->GetDrawingY();
+    double slope = 0.0;
+
+    /************** parent layers **************/
+
+    Note *note1 = dynamic_cast<Note *>(gliss->GetStart());
+    Note *note2 = dynamic_cast<Note *>(gliss->GetEnd());
+
+    if (!note1 && !note2) {
+        // no note, obviously nothing to do...
+        // this also means that notes with tstamp events are not supported
+        return;
+    }
+
+    LayerElement *durElement = NULL;
+    Chord *parentChord1 = NULL;
+    Layer *layer1 = NULL;
+    if (note1) {
+        durElement = note1;
+        layer1 = dynamic_cast<Layer *>(note1->GetFirstAncestor(LAYER));
+        parentChord1 = note1->IsChordTone();
+    }
+    if (parentChord1) {
+        durElement = parentChord1;
+    }
+
+    if (note1 || note2) {
+        int firstLoc = note1->GetDrawingLoc();
+        int secondLoc = note2->GetDrawingLoc();
```

Comment:
This will not work when you render a gliss over a page break because you have no guarantee that is is instantiated.

Reply:
What should be done?

Reply:
I think the only case where this is actually going to be problematic is when using encoded breaks. With '--breaks auto', I assume that the drawing loc will be instantiated for notes on the preceding and following page. You can test it an see what you get. It is something we need to solve for other cases. For example, determining the tie or the slur position. I think the best and simplest way to do this is to render the full document on one page before applying the encoded cast off. This is going to make it a bit slower but not any slower than '--breaks auto'

Reply:
I had several test runs with all possible option for `breaks` without any problems. Tried directly from MusicXML as well as converting to MEI and run it again. even multiple glissandi on a system / page break do not make any problems. 

Here's the test file:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.1 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="3.1">
  <work>
    <work-title>Glissandi with breaks</work-title>
    </work>
  <identification>
    <creator type="composer">Klaus Rettinghaus</creator>
    <encoding>
      <software>MuseScore 3.4.2</software>
      <encoding-date>2020-04-22</encoding-date>
      <supports element="accidental" type="yes"/>
      <supports element="beam" type="yes"/>
      <supports element="print" attribute="new-page" type="yes" value="yes"/>
      <supports element="print" attribute="new-system" type="yes" value="yes"/>
      <supports element="stem" type="yes"/>
      </encoding>
    </identification>
  <defaults>
    <scaling>
      <millimeters>7.05556</millimeters>
      <tenths>40</tenths>
      </scaling>
    <page-layout>
      <page-height>1683.36</page-height>
      <page-width>1190.88</page-width>
      <page-margins type="even">
        <left-margin>56.6929</left-margin>
        <right-margin>56.6929</right-margin>
        <top-margin>56.6929</top-margin>
        <bottom-margin>113.386</bottom-margin>
        </page-margins>
      <page-margins type="odd">
        <left-margin>56.6929</left-margin>
        <right-margin>56.6929</right-margin>
        <top-margin>56.6929</top-margin>
        <bottom-margin>113.386</bottom-margin>
        </page-margins>
      </page-layout>
    <word-font font-family="FreeSerif" font-size="10"/>
    <lyric-font font-family="FreeSerif" font-size="11"/>
    </defaults>
  <part-list>
    <score-part id="P1">
      <part-name>Piano</part-name>
      <part-abbreviation>Pno.</part-abbreviation>
      <score-instrument id="P1-I1">
        <instrument-name>Piano</instrument-name>
        </score-instrument>
      <midi-device id="P1-I1" port="1"></midi-device>
      <midi-instrument id="P1-I1">
        <midi-channel>1</midi-channel>
        <midi-program>1</midi-program>
        <volume>78.7402</volume>
        <pan>0</pan>
        </midi-instrument>
      </score-part>
    </part-list>
  <part id="P1">
    <measure number="1" width="314.55">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <top-system-distance>70.00</top-system-distance>
          </system-layout>
        </print>
      <attributes>
        <divisions>1</divisions>
        <key>
          <fifths>0</fifths>
          </key>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
          </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
          </clef>
        </attributes>
      <note default-x="79.27" default-y="0.00">
        <pitch>
          <step>F</step>
          <octave>5</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="1" type="start" default-y="-2.02">gliss.</slide>
          </notations>
        </note>
      </measure>
    <measure number="2" width="254.54">
      <note default-x="10.00" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-38.08">gliss.</slide>
          <slide line-type="solid" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="3" width="253.87">
      <note default-x="13.32" default-y="0.00">
        <pitch>
          <step>F</step>
          <octave>5</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="1" type="start" default-y="-1.98">gliss.</slide>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="4" width="254.54">
      <note default-x="10.00" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-38.24">gliss.</slide>
          <slide line-type="solid" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="5" width="303.06">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <system-distance>150.00</system-distance>
          </system-layout>
        </print>
      <note default-x="58.17" default-y="0.00">
        <pitch>
          <step>F</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="1" type="start" default-y="-0.48"/>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="179.64" default-y="-25.00">
        <pitch>
          <step>A</step>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>up</stem>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-26.04">gliss.</slide>
          </notations>
        </note>
      </measure>
    <measure number="6" width="251.42">
      <note default-x="10.00" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="3" type="start" default-y="-38.01">gliss.</slide>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="7" width="271.60">
      <note default-x="10.36" default-y="0.00">
        <pitch>
          <step>F</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="3" type="stop"/>
          </notations>
        </note>
      <note default-x="140.00" default-y="-20.00">
        <pitch>
          <step>B</step>
          <alter>-1</alter>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <accidental>flat</accidental>
        <stem>down</stem>
        </note>
      </measure>
    <measure number="8" width="251.42">
      <note default-x="10.00" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-37.52">gliss.</slide>
          <slide line-type="solid" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="9" width="280.82">
      <print new-page="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <top-system-distance>70.00</top-system-distance>
          </system-layout>
        </print>
      <note>
        <rest/>
        <duration>4</duration>
        <voice>1</voice>
        </note>
      </measure>
    <measure number="10" width="272.27">
      <note default-x="13.80" default-y="55.00">
        <pitch>
          <step>C</step>
          <octave>7</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <glissando line-type="wavy" number="1" type="start" default-y="45.15">gliss.</glissando>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="142.05" default-y="-35.00">
        <pitch>
          <step>F</step>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>up</stem>
        <notations>
          <glissando line-type="wavy" number="2" type="start" default-y="-39.92">gliss.</glissando>
          <glissando line-type="wavy" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="11" width="273.14">
      <note default-x="13.80" default-y="-110.00">
        <pitch>
          <step>E</step>
          <octave>2</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>up</stem>
        <notations>
          <glissando line-type="wavy" number="1" type="start" default-y="-101.72">gliss.</glissando>
          <glissando line-type="wavy" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="142.49" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>up</stem>
        <notations>
          <glissando line-type="wavy" number="2" type="start" default-y="-29.00">gliss.</glissando>
          <glissando line-type="wavy" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="12" width="251.26">
      <note default-x="10.36" default-y="-15.00">
        <pitch>
          <step>C</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="1" type="start" default-y="-15.23">gliss.</slide>
          <glissando line-type="wavy" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="129.83" default-y="-10.00">
        <pitch>
          <step>D</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-9.42">gliss.</slide>
          </notations>
        </note>
      </measure>
    <measure number="13" width="301.27">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <system-distance>150.00</system-distance>
          </system-layout>
        </print>
      <note default-x="58.17" default-y="0.00">
        <pitch>
          <step>F</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="178.74" default-y="-10.00">
        <pitch>
          <step>D</step>
          <alter>1</alter>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <accidental>sharp</accidental>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="2" type="start" default-y="-10.68">gliss.</slide>
          </notations>
        </note>
      </measure>
    <measure number="14" width="259.29">
      <note default-x="16.63" default-y="-20.00">
        <pitch>
          <step>B</step>
          <alter>-1</alter>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <accidental>flat</accidental>
        <stem>down</stem>
        <notations>
          <slide line-type="solid" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="136.98" default-y="-30.00">
        <pitch>
          <step>G</step>
          <alter>1</alter>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <accidental>sharp</accidental>
        <stem>up</stem>
        <notations>
          <slide line-type="solid" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="15" width="273.42">
      <note default-x="10.36" default-y="-5.00">
        <pitch>
          <step>E</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <glissando line-type="wavy" number="1" type="start" default-y="-2.83">gliss.</glissando>
          </notations>
        </note>
      <note default-x="140.91" default-y="25.00">
        <pitch>
          <step>D</step>
          <alter>1</alter>
          <octave>6</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <accidental>sharp</accidental>
        <stem>down</stem>
        <notations>
          <glissando line-type="wavy" number="2" type="start" default-y="21.56">gliss.</glissando>
          <glissando line-type="wavy" number="1" type="stop"/>
          </notations>
        </note>
      </measure>
    <measure number="16" width="243.52">
      <note default-x="10.36" default-y="-10.00">
        <pitch>
          <step>D</step>
          <octave>5</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>down</stem>
        <notations>
          <glissando line-type="wavy" number="2" type="stop"/>
          </notations>
        </note>
      <note default-x="121.46" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>half</type>
        <stem>up</stem>
        <notations>
          <fermata type="upright"/>
          </notations>
        </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
        </barline>
      </measure>
    </part>
  </score-partwise>
```